teleprox/proxy.py

```diff
@@ -223,56 +228,77 @@ def _set_proxy_options(self, **kwds):
         """
         for k in kwds:
             if k not in self._proxy_options:
-                raise KeyError("Unrecognized proxy option '%s'" % k)
+                raise KeyError(f"Unrecognized proxy option '{k}'")
         self._proxy_options.update(kwds)
 
     def _get_value(self):
         """
         Return the value of the proxied object.
-        
+
         If the object is not serializable, then raise an exception.
         """
         if self._client() is None:
-            return self._server().unwrap_proxy(self)
-        else:
+            return self._local_server.unwrap_proxy(self)
+        try:
             return self._client().get_obj(self, return_type='value')
+        except TimeoutError as exc:
+            if self._server_is_lazy:
+                exc.add_note(
+                    "This proxy object is being accessed synchronously when the server it references "
```

Comment:
maybe -- what if the server side actually is doing lazy processing? I think we can't tell from here..

Reply:
(same for more instances of the same code below)